Storage/src/Connection/RetryTrait.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Storage\Connection;
+
+/**
+ * Trait which provides helper methods for retry logic.
+ */
+trait RetryTrait
+{
+    /**
+     * @var array
+     */
+    private $httpRetryCodes = [
+        0, // connetion-refused OR connection-reset gives status code of 0
+        408,
+        429,
+        500,
+        502,
+        503,
+        504
+    ];
+
+    /**
+     * @var array
+     */
+    private $opsDescriptionMap = [
+        'bucket_acl.get' => 'idempotent',
```

Comment:
Again not an expert on PHP, but could you make idempotent/idempotent_with_precondition into consts or an enum type rather than repeated strings? Or what @saranshdhingra suggests below sounds like it would make this cleaner also.

Reply:
So, consts in traits are available from PHP 8.2, enums as a whole recently became available.
The best we can do is static variables, but even those(AFAIK) we can't use as array values(when declaring as a property).

But let me see how we can approach an even cleaner solution.